src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/AccessToken/OidcUserInfoTokenHandlerFactory.php

```diff
@@ -61,14 +45,9 @@ public function addConfiguration(NodeBuilder $node): void
                         ->info('Claim which contains the user identifier (e.g.: sub, email..).')
                         ->defaultValue('sub')
                     ->end()
-                    ->arrayNode('client')
+                    ->scalarNode('client')
                         ->info('HttpClient to call the OIDC server.')
                         ->isRequired()
```

Comment:
why do we keep this? can't we default to `http_client`?

Reply:
Cause we need the base_uri of the OIDC server:
```yaml
# config/packages/framework.yaml
framework:
    http_client:
        scoped_clients:
            oidc.client:
                base_uri: 'https://www.example.com/realms/demo/protocol/openid-connect/userinfo'

# config/packages/security.yaml
security:
    firewalls:
        main:
            access_token:
                token_handler:
                    oidc_user_info:
                        client: oidc.client
```

Reply:
are there any other options that might be useful to configure?
because if not, we might just need oidc_user_info: <base-uri-as-string>
When using the withOptions as a factory to create the client you need
To do so, I'd suggest defining an abstract service, let's name it `security.access_token_handler.oidc_user_info.http_client`, defined as such:

```php
->set('security.access_token_handler.oidc_user_info', HttpClientInterface::class)
	->abstract()
	->factory([service('http_client'), 'withOptions'])
```

and in the OidcUserInfoTokenHandlerFactory, we'd create a ChildDefinition of this service and set it's argument to `['base_uri' => $config['base_uri']]`.

Does this make sense to you?

Reply:
`oidc_user_info` can also accept a `claim` optional parameter:
```yaml
oidc_user_info:
  claim: email # default: sub
  client: oidc.client # required
```

I can allow `oidc_user_info: <base-uri>` and normalize it, but I find this syntax more confusing than setting the `oidc_user_info.client` parameter. But this is just a question of point of view. WDYT?

Reply:
We should keep the `client` option in case the dev wants to manually configure its client externally (e.g.: `framework.http_client`, or any service manually created).

In your approach, the `base_uri` option would be used to dynamically create the client (and set the `client` option), and the `client` option would receive a HttpClient service id:
```yaml
# HttpClient dynamically created with "base_uri" only
oidc_user_info: https://...

# HttpClient dynamically created with "base_uri" only
oidc_user_info:
  claim: sub
  base_uri: https://... # "base_uri" is not compatible with "client" option

# HttpClient manually created with custom options
oidc_user_info:
  claim: sub
  client: oidc.client
```

But I think it's making this configuration and DX more complex... IMHO, I would prefer to set the HttpClient options in the `client` option like it was earlier (but this time, using the abstract definition you suggested):
```yaml
oidc_user_info:
  claim: sub
  client:
    base_uri: https://...
    # other HttpClient options
```

Reply:
I'm fine adding another option to set the service id if that's something that'd be useful, but I don't thing allowing other http_client options is desired (as explained in https://github.com/symfony/symfony/pull/50453#discussion_r1210028968)
But being able to set only the base_uri looks the best DX to me. I can do the code if you're not confortable with it of course. Let me know!